packages/source-fetcher/lib/etherscan.ts

```diff
@@ -86,12 +89,24 @@ const EtherscanFetcher: FetcherConstructor = class EtherscanFetcher
       { retries: 3, minTimeout: this.delay * initialTimeoutFactor }
     );
   }
-
+  
+  private urlSwitcher(address: string){
```

Comment:
This function needs to be renamed.  The name `urlSwitcher` does not describe what it does.  Usually, functions are named as verb phrases.  I would suggest a name like `determineUrl`.

Also, you've given this function a parameter `address`, but it's never used.  That should be removed.

Also I notice you've gone the route of making this an instance method, and having it switch based on `this.networkName`...  that works, but maybe it would be better to have `networkName` passed in explicitly as a parameter, and have this function be a static method (or outside the class entirely) rather than an instance method?  I feel like that would make the data flow more explicit.  Pure functions are generally preferable when feasible.

(In fact, down below, where you actually *invoke* it, you invoked it as if the network name was a parameter to be passed in!  Whichever route you go, better make it consistent.)

Also, this function is missing a return signature; you should add that.

Also -- sorry for more of these tiny things that should be handled by `prettier`, hopefully @cds-amal will have that fixed soon -- but there should be a space between `)` and `{`.  (You have this as well on the `switch` line.)

Reply:
This seems to have gotten marked resolved, but I'm marking it unresolved because not all of the above has been addressed.

Reply:
Cool we updated this function and changed the name to `determineUrl`. Thanks for the catch.

Reply:
Cool, I see also you took the approach of making it an instance method rather than a pure function.  That works -- as long as you're consistent. :)